lang/py/avro/schema.py

```diff
@@ -1069,6 +1070,33 @@ def validate(self, datum):
     def __eq__(self, that):
         return self.props == that.props
 
+
+#
+# uuid Type
+#
+
+
+class UUIDSchema(LogicalSchema, PrimitiveSchema):
+    def __init__(self, other_props=None):
+        LogicalSchema.__init__(self, avro.constants.UUID)
+        PrimitiveSchema.__init__(self, 'string', other_props)
+
+    def to_json(self, names=None):
+        return self.props
+
+    def validate(self, datum):
+        try:
+            val = uuid.UUID(datum, version=4)
+        except ValueError:
+            # If it's a value error, then the string
+            # is not a valid hex code for a UUID.
+            return False
```

Comment:
```suggestion
            return None
```
for consistency with the others

Reply:
Handled as part of https://github.com/apache/avro/pull/1148/commits/f7ec8498193cbd2d853489aedefdcbedc86315a4